src/helpers.ts

```diff
@@ -23,7 +23,12 @@ export function normalizeRecord(r: { [key: string]: unknown } | Result): { [key:
   return Object.fromEntries(
     Object.keys(r)
       .filter((k) => isNaN(Number(k)))
-      .map((k) => [k, r[k]])
+      .map((k) => {
+        if (r[k] instanceof BigNumber) {
+          return [k, r[k].toString()];
```

Comment:
I'd take this further by doing this:

```
const floatThreshold = parseUnits("1", 18);
if (r[k].gte(floatThreshold)) {
  return [k, formatUnits(r[k], 18)];
}
```

Reply:
This will overflow. 1.0 tokens = 1^18 bignumber.

Reply:
The edited version should work.

Reply:
It works, but you can't apply it unconditionally to all bignumbers since some of them aren't fractional (e.g. 1 == 1, but 1.0 = 1^18).